lib/swig/SharedLibraryModel.h

```diff
@@ -0,0 +1,322 @@
+#else
+        void *symbol = dlsym(m_Library, symbolName.c_str());
+#endif
+
+        // If this symbol isn't allowed to be missing but it is, raise exception
+        if(!allowMissing && symbol == nullptr) {
+            throw std::runtime_error("Cannot find symbol '" + symbolName + "'");
+        }
+
+        return symbol;
+    }
+
+    void allocateMem()
+    {
+        m_AllocateMem();
+    }
+
+    void allocateSparsePop( const std::string &popName, unsigned int nConn )
+    {
+      typedef void(*UIntFct)(unsigned int);
+      ((UIntFct) getSymbol( "allocate" + popName ))( nConn );
+    }
+
+    void allocateExtraGlobalParam( const std::string &popName,
+                                   const std::string &paramName,
+                                   const int size )
+    {
+        auto egp = static_cast<void**>(getSymbol( paramName + popName ));
+#ifndef CPU_ONLY
+        CHECK_CUDA_ERRORS( cudaHostAlloc( egp, size * sizeof( scalar ), cudaHostAllocPortable ) );
+#else
+        *egp = malloc( size * sizeof( scalar ) );
+#endif
+    }
+
+    void freeExtraGlobalParam( const std::string &popName,
+                               const std::string &paramName )
+    {
+        auto egp = static_cast<void**>( getSymbol( paramName + popName ));
+#ifndef CPU_ONLY
+        CHECK_CUDA_ERRORS( cudaFreeHost( egp ) );
+#else
+        free(*egp);
+#endif
+    }
+
+    void initialize()
+    {
+        m_Initialize();
+    }
+
+    void initializeSparsePop( const std::string &popName,
+                              unsigned int* _ind, int nConn,
+                              unsigned int* _indInG, int nPre,
+                              scalar* _g, int nG )
+    {
+        auto C = static_cast<SparseProjection*>( getSymbol( "C" + popName ) );
+        auto g = static_cast<scalar**>( getSymbol( "g" + popName ) );
+        C->connN = nConn;
+        for ( int i = 0; i < nConn; ++i )
+        {
+            C->ind[i] = _ind[i];
+        }
+        for ( int i = 0; i < nPre; ++i )
+        {
+            C->indInG[i] = _indInG[i];
+        }
+        for ( int i = 0; i < nG; ++i )
+        {
+            (*g)[i] = _g[i];
+        }
+    }
+
+    void initializeModel()
+    {
+        m_InitializeModel();
+    }
+
+    void stepTimeGPU()
+    {
+        m_StepTimeGPU();
+    }
+
+    void stepTimeCPU()
+    {
+        m_StepTimeCPU();
+    }
+
+    void exitGeNN()
+    {
+        m_ExitGeNN();
+    }
+
+private:
+    //----------------------------------------------------------------------------
+    // Members
+    //----------------------------------------------------------------------------
+#ifdef _WIN32
+    HMODULE m_Library;
+#else
+    void *m_Library;
+#endif
+
+    VoidFunction m_AllocateMem;
+    VoidFunction m_Initialize;
+    VoidFunction m_InitializeModel;
+    VoidFunction m_StepTimeGPU;
+    VoidFunction m_StepTimeCPU;
+    VoidFunction m_ExitGeNN;
+    
+    PopIOMap populationsIO;
```

Comment:
This should be ``m_PopulationsIO`` in keeping with the general naming convention.